src/Npgsql/NpgsqlRawCopyStream.cs

```diff
@@ -487,6 +487,8 @@ public void Cancel()
     /// Cancels and terminates an ongoing import. Any data already written will be discarded.
     /// </summary>
     public Task CancelAsync() => ((NpgsqlRawCopyStream)BaseStream).CancelAsync();
+
+    public override ValueTask DisposeAsync() => ((NpgsqlRawCopyStream)BaseStream).DisposeAsync();
```

Comment:
Wait, doesn't `StreamWriter` already implement `DisposeAsync`?

Reply:
Yes but only explicitly, they call into Dispose otherwise (they don't even try to do a `this is IAsyncDisposable`)

Reply:
On the other hand `StreamWriter` has its own writer which is flushed on dispose...

Reply:
I don't think it's really worth to stop synchronously blocking on disposing `NpgsqlCopyTextWriter` just on the chance we introduce more bugs.

Reply:
Yeah right, we should flush as well.

The reason this came up is because I noticed sync reads while I wasn't expecting them.

Reply:
It seems pretty hellish to get this to work properly yeah. even FlushAsync and friends are guarded by these checks... Wonder what their intent was.

Reply:
Wrote a bit on https://github.com/dotnet/runtime/issues/88246#issuecomment-1828460404